src/zenml/integrations/feast/steps/feast_feature_store_step.py

```diff
@@ -0,0 +1,134 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from typing import Any, Dict, List, Optional, Type, Union
+
+from feast.feature_service import FeatureService  # type: ignore[import]
+from pandas import DataFrame
+
+from zenml.steps import BaseStep, BaseStepConfig, StepContext, step
+
+
+class FeastHistoricalFeaturesConfig(BaseStepConfig):
```

Comment:
This and the `FeastOnlineFeaturesConfig` seem really weird to me. Would it not make more sense to pass the parameters you pass in the functions via these configs?

E.g. 

```python
def feast_historical_features_step(
    entity_df: Union[DataFrame, str],
    features: Union[List[str], FeatureService],
    full_feature_names: bool = False,
    name: Optional[str] = None,
    enable_cache: Optional[bool] = None,
) -> Type[BaseStep]:
```

could rather be: 

```python
class FeastHistoricalFeaturesConfig(BaseStepConfig):
    entity_df: Union[DataFrame, str],
    features: Union[List[str], FeatureService],
    full_feature_names: bool = False,

def feast_historical_features_step(name, enable_cache, config: FeastHistoricalFeaturesConfig)
```

Reply:
Also, on second look, we do not need a function within a function. We can simply expose the inner function directly.

Reply:
The problem with passing them in via the Config is that we are then passing in DataFrames through the Config, which means I have to write custom serialisation logic so that Pydantic can serialise the data, and so it can be added to our metadata store. Not sure I'll get that working by the end of today if that's what we choose to do.

Reply:
This is handled now.